File: app/src/main/java/com/geniusforapp/movies/ui/activites/BaseActivity.java
package com.geniusforapp.movies.ui.activites;

import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import com.geniusforapp.movies.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by anajar on 5/14/17.
 */

public class BaseActivity extends AppCompatActivity {


    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @Override
    public void setContentView(@LayoutRes int layoutResID) {
        super.setContentView(layoutResID);
        bindViews();
    }

    public void setContentViewWithoutInjections(@LayoutRes int layoutResID) {
        super.setContentView(layoutResID);

    }


    protected void bindViews() {
        ButterKnife.bind(this);
        setupToolBar();
    }

    protected void setupToolBar() {
        if (toolbar != null) {
            setSupportActionBar(toolbar);
        }
    }

    public void showBack() {
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }
}

File: app/src/main/java/com/geniusforapp/movies/ui/base/BaseActivity.kt
package com.geniusforapp.movies.ui.base

import android.os.Bundle
import android.preference.PreferenceManager
import android.view.MenuItem
import androidx.annotation.LayoutRes
import com.geniusforapp.movies.R
import dagger.android.support.DaggerAppCompatActivity
import kotlinx.android.synthetic.main.layout_toolbar.*

/**
 * Created by anajar on 5/14/17.
 */

abstract class BaseActivity : DaggerAppCompatActivity() {

    private var isBackArrowEnabled: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(PreferenceManager.getDefaultSharedPreferences(this).getInt("theme", R.style.AppTheme))
        super.onCreate(savedInstanceState)
    }

    override fun setContentView(@LayoutRes layoutResID: Int) {
        super.setContentView(layoutResID)
        bindViews()
    }

    private fun bindViews() {
        setupToolBar()
    }

    private fun setupToolBar() {
        if (toolbar != null) {
            setSupportActionBar(toolbar)
        }
    }

    fun showBack() {
        isBackArrowEnabled = true
        supportActionBar?.setDisplayHomeAsUpEnabled(isBackArrowEnabled)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            // Respond to the action bar's Up/Home button
            android.R.id.home -> {
                if (isBackArrowEnabled) {
                    onBackPressed()
                }
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

}
